Add state diagram

fwolter opened the pull request:
I used the already present approach to draw the diagram. I think this fits better the project. My intention when drawing this was to repesent the current implementation.

I made a fat version to understand when VDR calls Clear() and a boiled down version, merging all trick speed states. If I see correctly they can all be handled the same as VDR takes care to call Clear() in the right moment.

fwolter committed:
Add state diagram

rellla commented:
I will have look later. One thing... SetPlayMode() is used to Start and Stop Live streams, while Play/Freeze/Trickspeed is used to handle playback in recordings. So SetPlayMode != Play and doesn't call it at all.

If you want to make it complete, you can add the SkipSeconds path (green/yellow key jump) and Stillpicture (jumping to marks). I really had only a quick look at the diagram.

fwolter committed:
Move the simple version to the top. Add SkipSeconds..

fwolter commented:
When I start playing back a recording, I get `[softhddevice] device: SetPlayMode: 1`, and no Play(). I couldn't test it with live TV, yet. Is it that what you mean by live stream?

I already wondered when StillPicture() is used. Added!

I added SkipSeconds only to the fat version, because it doesn't appear to need explicit handling by the plugin.

rellla commented:
> When I start playing back a recording, I get `[softhddevice] device: SetPlayMode: 1`, and no Play(). I couldn't test it with live TV, yet. Is it that what you mean by live stream?
> 
> I already wondered when StillPicture() is used. Added!
> 
> I added SkipSeconds only to the fat version, because it doesn't appear to need explicit handling by the plugin.

Ok, then maybe i was wrong. I can't test it atm, but Play() is then used when coming from freeze or trickspeed probably.

fwolter commented:
Yes, it is!

fwolter committed:
Add missing cancel filter thread in PrepareStop

rellla commented:
What do you think of adding https://github.com/rellla/vdr-plugin-softhddevice-drm-gles/commit/a4aabec618e34731354807c2874a64891ab74cf8 in addition to your graphs?
This is a simple graph of my table without anything else.

fwolter commented:
Sure, looks good! Every sight which contributes to get a clearer view of how things connect/flow helps!

rellla commented:
One should never never never rebase the master branch, but i did it. Sorry.

fwolter commented:
On what did you rebase the main branch? I don't see any impact.

rellla commented:
> On what did you rebase the main branch? I don't see any impact.

I was fast ;) If you haven't pulled directly after the last two merges, you won't get any notice. I just forgot to squash the single commits in the diagram PRs.
